app/helpers/search_helper.rb

```diff
@@ -0,0 +1,230 @@
+  end
+
+  # these images have been resized with
+  # mogrify -negate -alpha copy -negate -resize 42x42 -background transparent -gravity center -extent 42x42 *.png
+  # or
+  # mogrify -negate -alpha copy -negate -resize 72x72 -background transparent -gravity center -extent 72x72 *.png
+
+  # Sign Attribute is Selected?
+
+  def handshape_selected?(shape)
+    return if @query[:hs].blank?
+
+    query_hs = @query[:hs]
+
+    # if it's the first, the search is just on the first two numbers
+    query_hs = @query[:hs].map { |q| "#{q}.1" } if shape.split('.').last == '1'
+
+    'selected' if query_hs.include?(shape)
+  end
+
+  def location_selected?(location)
+    'selected' if @query[:l].present? && @query[:l].include?(location.split('.')[1])
+  end
+
+  def location_group_selected?(location_group)
+    'selected' if @query[:lg].present? && @query[:lg].include?(location_group.split('.')[0])
+  end
+
+  def tab_class(*classes)
+    selected = tab_selected?(classes) if params[:tab].blank?
+    # note: comparing as a string, to avoid a DOS ruby bug
+    #   # see http://brakemanscanner.org/docs/warning_types/denial_of_service/
+    selected = true if classes.map(&:to_s).include?(params[:tab])
+
+    classes << (selected ? :selected : '')
+    classes.join(' ')
+  end
+
+  def tab_classes
+    [
+      tab_class(:advanced),
+      'clearfix',
+      'medium-10',
+      'medium-centered',
+      'large-8',
+      'advanced-search-container',
+      'show-for-medium'
+    ]
+  end
+
+  def tab_selected?(classes)
+    keys = @query.select { |_key, value| value.present? }.keys
+
+    if %w(tag usage).any? { |key| keys.include?(key) } || (keys.include?('s') && keys.length > 1)
+      classes.include?(:advanced)
+    elsif %w(hs l lg).any? { |key| keys.include?(key) }
+      classes.include?(:signs)
+    else
+      classes.include?(:keywords)
+    end
+  end
+
+  def display_locations_search_term(simple = false)
+    # reduce the list to the selected, turn them all into images.
+    locations = ::SignMenu.resolve.locations.flatten.select do |location|
+      location_selected?(location)
+    end
+    return if @query[:l].blank?
+
+    locations = locations.map do |location|
+      location_image(
+        location,
+        false,
+        false,
+        simple
+      )
+    end
+
+    safe_join(locations, ' ')
+  end
+
+  def display_handshapes_search_term(simple = false)
+    selected = ::SignMenu.resolve.handshapes.flatten.flatten.select do |hand_shape|
+      handshape_selected?(hand_shape)
+    end
+    return if @query[:hs].blank?
+
+    selected = selected.map do |hand_shape|
+      handshape_image(
+        hand_shape,
+        hand_shape.split('.').last == '1',
+        simple
+      )
+    end
+
+    safe_join(selected, ' ')
+  end
+
+  def display_location_groups_search_term(simple = false)
+    locations = ::SignMenu.resolve.location_groups.select do |location_group|
```

Comment:
@joshmcarthur is `::` syntactic sugar when used like this?

Reply:
This is leftover from when `SearchHelper` was `Freelex::SearchHelper`. Prefixing a class or module name with `::` causes Ruby to check the top-level scope for the class, instead of the module.

Because we have a class named `Freelex::SignMenu` (which the top-level `SignMenu#resolve` method resolves to when the app is in Freelex dictionary mode), the `::` was necessary when `SearchHelper` was `Freelex::SearchHelper` to ensure that Ruby would find the correct classs.